public/data.js

```diff
@@ -27,19 +27,37 @@ class DataController {
             // Update data every time it changes in firestore
             DataController.updateData(res, activeProjects);
             onChange(activeProjects);
+        }, error => {
+            if (error.code == "permission-denied") {
+                firebase.auth().onAuthStateChanged(user => { 
+                    if (user) {
+                        name = user.displayName.split(" ").slice(0,-1)
+                        // Error message - Alert Missing or insufficient permissions.
+                        alert(`${error.message} 
+                        \r Please ${name} use AVF email or liaise with Africa's Voices Foundation (AVF) to give you AVF email`) 
+                        if (user.email.match(".*@africasvoices.org$")) {
```

Comment:
This still locks out non -avf but authorized email addresses

Reply:
Please note that the logic is after the error permission-denied (The user has been already locked out at this point) and that the logic is used to generate feedback that would be relevant to the user.